### PR TITLE
Refactored fmt output to loginfo output

### PR DESCRIPTION
**Pull request by mattyzzy**
Some of the other comment line is remained for now because I need to discuss some more thing with Geoff about how to implement disconnect in auto paho. They will be resolved as soon as the new version is implemented.

### CODE REVIEW

File: common/drivers/edgeConnectorMQTT.go [hunk 1]
```diff
@@ -135,20 +135,20 @@ func (s *edgeConnectorMQTT) Connect(clientId string) error {
 		KeepAlive:         300,
 		ConnectRetryDelay: 10 * time.Second,
 		OnConnectionUp: func(cm *autopaho.ConnectionManager, connAck *paho.Connack) {
-			fmt.Println("mqtt connection up")
+			s.LogInfo("mqtt connection up")
 		},
-		OnConnectError: func(err error) { fmt.Printf("error whilst attempting connection: %s\n", err) },
+		OnConnectError: func(err error) { s.LogInfo("error whilst attempting connection: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/edgeConnectorMQTT.go:140]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/edgeConnectorMQTT.go [hunk 2]
```diff
@@ -135,20 +135,20 @@ func (s *edgeConnectorMQTT) Connect(clientId string) error {
...
 		ClientConfig: paho.ClientConfig{
 			ClientID: clientId,
 			Router: paho.NewSingleHandlerRouter(func(m *paho.Publish) {
 				s.tagChangeHandler(m)
 			}),
-			OnClientError: func(err error) { fmt.Printf("server requested disconnect: %s\n", err) },
+			OnClientError: func(err error) { s.LogInfo("server requested disconnect: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/edgeConnectorMQTT.go:146]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/libreConnectorMQTT.go [hunk 3]
```diff
@@ -117,18 +116,17 @@ func (s *libreConnectorMQTT) Connect() error {
 		KeepAlive:         300,
 		ConnectRetryDelay: 10 * time.Second,
 		OnConnectionUp: func(cm *autopaho.ConnectionManager, connAck *paho.Connack) {
-			fmt.Println("mqtt connection up")
+			s.LogInfo("mqtt connection up")
 		},
-		OnConnectError: func(err error) { fmt.Printf("error whilst attempting connection: %s\n", err) },
+		OnConnectError: func(err error) { s.LogInfo("error whilst attempting connection: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/libreConnectorMQTT.go:121]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/libreConnectorMQTT.go [hunk 4]
```diff
@@ -117,18 +116,17 @@ func (s *libreConnectorMQTT) Connect() error {
...
 		ClientConfig: paho.ClientConfig{
 			ClientID: svcName,
-			//TODO: no router?
-			OnClientError: func(err error) { fmt.Printf("server requested disconnect: %s\n", err) },
+			OnClientError: func(err error) { s.LogInfo("server requested disconnect: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/libreConnectorMQTT.go:124]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/plcConnectorMQTT.go [hunk 5]
```diff
@@ -87,20 +88,20 @@ func (s *plcConnectorMQTT) Connect() error {
 		KeepAlive:         300,
 		ConnectRetryDelay: 10 * time.Second,
 		OnConnectionUp: func(cm *autopaho.ConnectionManager, connAck *paho.Connack) {
-			fmt.Println("mqtt connection up")
+			s.LogInfo("mqtt connection up")
 		},
-		OnConnectError: func(err error) { fmt.Printf("error whilst attempting connection: %s\n", err) },
+		OnConnectError: func(err error) { s.LogInfo("error whilst attempting connection: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/plcConnectorMQTT.go:93]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/plcConnectorMQTT.go [hunk 6]
```diff
@@ -87,20 +88,20 @@ func (s *plcConnectorMQTT) Connect() error {
...
 		ClientConfig: paho.ClientConfig{
 			ClientID: svcName,
 			Router: paho.NewSingleHandlerRouter(func(m *paho.Publish) {
 				s.receivedMessageHandler(m)
 			}),
-			OnClientError: func(err error) { fmt.Printf("server requested disconnect: %s\n", err) },
+			OnClientError: func(err error) { s.LogInfo("server requested disconnect: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/plcConnectorMQTT.go:99]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/plcConnectorMQTT.go [hunk 7]
```diff
@@ -87,20 +88,20 @@ func (s *plcConnectorMQTT) Connect() error {
...
 			OnServerDisconnect: func(d *paho.Disconnect) {
 				if d.Properties != nil {
-					fmt.Printf("server requested disconnect: %s\n", d.Properties.ReasonString)
+					s.LogInfo("server requested disconnect: %s\n", d.Properties.ReasonString)
```
Comment (tomhollingworth) [inline common/drivers/plcConnectorMQTT.go:102]: Maybe s.LogWarn is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/plcConnectorMQTT.go [hunk 8]
```diff
@@ -87,20 +88,20 @@ func (s *plcConnectorMQTT) Connect() error {
...
 				} else {
-					fmt.Printf("server requested disconnect; reason code: %d\n", d.ReasonCode)
+					s.LogInfo("server requested disconnect; reason code: %d\n", d.ReasonCode)
```
Comment (tomhollingworth) [inline common/drivers/plcConnectorMQTT.go:104]: Maybe s.LogWarn is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/edgeConnectorMQTT.go [hunk 9]
```diff
@@ -135,20 +135,20 @@ func (s *edgeConnectorMQTT) Connect(clientId string) error {
...
 			OnServerDisconnect: func(d *paho.Disconnect) {
 				if d.Properties != nil {
-					fmt.Printf("server requested disconnect: %s\n", d.Properties.ReasonString)
+					s.LogInfo("server requested disconnect: %s\n", d.Properties.ReasonString)
```
Comment (tomhollingworth) [inline common/drivers/edgeConnectorMQTT.go:149]: Maybe s.LogWarn is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/edgeConnectorMQTT.go [hunk 10]
```diff
@@ -135,20 +135,20 @@ func (s *edgeConnectorMQTT) Connect(clientId string) error {
...
 				} else {
-					fmt.Printf("server requested disconnect; reason code: %d\n", d.ReasonCode)
+					s.LogInfo("server requested disconnect; reason code: %d\n", d.ReasonCode)
```
Comment (tomhollingworth) [inline common/drivers/edgeConnectorMQTT.go:151]: Maybe s.LogWarn is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/pubSubConnectorMQTT.go [hunk 11]
```diff
@@ -108,20 +107,20 @@ func (s *pubSubConnectorMQTT) Connect() error {
...
 			}),
-			OnClientError: func(err error) { fmt.Printf("server requested disconnect: %s\n", err) },
+			OnClientError: func(err error) { s.LogInfo("server requested disconnect: %s\n", err) },
 			OnServerDisconnect: func(d *paho.Disconnect) {
 				if d.Properties != nil {
-					fmt.Printf("server requested disconnect: %s\n", d.Properties.ReasonString)
+					s.LogInfo("server requested disconnect: %s\n", d.Properties.ReasonString)
 				} else {
-					fmt.Printf("server requested disconnect; reason code: %d\n", d.ReasonCode)
+					s.LogInfo("server requested disconnect; reason code: %d\n", d.ReasonCode)
```
Comment (tomhollingworth) [inline common/drivers/pubSubConnectorMQTT.go:123]: Maybe s.LogWarn is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/pubSubConnectorMQTT.go [hunk 12]
```diff
@@ -108,20 +107,20 @@ func (s *pubSubConnectorMQTT) Connect() error {
...
 			ClientID: svcName,
 			Router: paho.NewSingleHandlerRouter(func(m *paho.Publish) {
 				s.tagChangeHandler(m)
 			}),
-			OnClientError: func(err error) { fmt.Printf("server requested disconnect: %s\n", err) },
+			OnClientError: func(err error) { s.LogInfo("server requested disconnect: %s\n", err) },
 			OnServerDisconnect: func(d *paho.Disconnect) {
 				if d.Properties != nil {
-					fmt.Printf("server requested disconnect: %s\n", d.Properties.ReasonString)
+					s.LogInfo("server requested disconnect: %s\n", d.Properties.ReasonString)
```
Comment (tomhollingworth) [inline common/drivers/pubSubConnectorMQTT.go:121]: Maybe s.LogWarn is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/pubSubConnectorMQTT.go [hunk 13]
```diff
@@ -108,20 +107,20 @@ func (s *pubSubConnectorMQTT) Connect() error {
...
 		},
-		OnConnectError: func(err error) { fmt.Printf("error whilst attempting connection: %s\n", err) },
+		OnConnectError: func(err error) { s.LogInfo("error whilst attempting connection: %s\n", err) },
 		ClientConfig: paho.ClientConfig{
 			ClientID: svcName,
 			Router: paho.NewSingleHandlerRouter(func(m *paho.Publish) {
 				s.tagChangeHandler(m)
 			}),
-			OnClientError: func(err error) { fmt.Printf("server requested disconnect: %s\n", err) },
+			OnClientError: func(err error) { s.LogInfo("server requested disconnect: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/pubSubConnectorMQTT.go:118]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

File: common/drivers/pubSubConnectorMQTT.go [hunk 14]
```diff
@@ -108,20 +107,20 @@ func (s *pubSubConnectorMQTT) Connect() error {
 		KeepAlive:         300,
 		ConnectRetryDelay: 10 * time.Second,
 		OnConnectionUp: func(cm *autopaho.ConnectionManager, connAck *paho.Connack) {
-			fmt.Println("mqtt connection up")
+			s.LogInfo("mqtt connection up")
 		},
-		OnConnectError: func(err error) { fmt.Printf("error whilst attempting connection: %s\n", err) },
+		OnConnectError: func(err error) { s.LogInfo("error whilst attempting connection: %s\n", err) },
```
Comment (tomhollingworth) [inline common/drivers/pubSubConnectorMQTT.go:112]: Maybe s.LogError is better given the context. Sorry I should of taken a closer look at the time.

**Review by tomhollingworth — CHANGES_REQUESTED** [review]
Hey @mattyzzy I was a bit lazy in the previous review and didn't properly check the context of the logging. Can we use s.logWarn / s.logError where appropriate. Happy for you to make your own judgement on which type of logger to use

File: common/drivers/edgeConnectorMQTT.go (hunk 1 above)
Comment (mattyzzy) [reply common/drivers/edgeConnectorMQTT.go:140]: Hi @tomhollingworth I had a few seconds of hesitation when I reaad your comment as well but using different loggers absolutely makes more sense to me! Thanks for letting me know! I'll make these changes tomorrow.

**Review by tomhollingworth — APPROVED** [review]
Awesome, thanks @mattyzzy

**Pull request merged**